src/proxy/MultiMap.ts

```diff
@@ -0,0 +1,172 @@
+import Promise = Q.Promise;
+import {DistributedObject} from '../DistributedObject';
+import {IMapListener} from '../core/MapListener';
+
+export interface MultiMap<K, V> extends DistributedObject {
+
+    /**
+     * Adds a key-value pair to this multi-map.
+     * If this multi-map already has some value associated with the specified key, then calling this method
+     * will not replace the old value. Instead, both values will be associated with the same key.
+     * @param key key to add.
+     * @param value value to associate with the key.
+     * @return `true` if this multi-map did not have the specified value associated
+     * with the specified key, `false` otherwise.
+     */
+    put(key: K, value: V): Promise<boolean>;
+
+    /**
+     * Retrieves a list of values associated with the specified key.
+     * @param key key to search for.
+     * @return array of values associated with the specified key.
+     */
+    get(key: K): Promise<Array<V>>;
+
+    /**
+     * Removes an association of the specified value with the specified key. Calling this method does not affect
+     * other values associated with the same key.
+     * @param key key from which the value should be detached.
+     * @param value value to be removed.
+     * @return `true` if the value was detached from the specified key, `false` if it was not.
+     */
+    remove(key: K, value: V): Promise<boolean>;
+
+    /**
+     * Detaches all values from the specified key.
+     * @param key key from which all entries should be removed.
+     * @return an array of old values that were associated with this key prior to this method call.
+     */
+    removeAll(key: K): Promise<Array<V>>;
+
+    /**
+     * @return an array of all keys in this multi-map.
+     */
+    keySet(): Promise<Array<K>>;
+
+    /**
+     * @return a flat array of all values stored in this multi-map.
+     */
+    values(): Promise<Array<V>>;
```

Comment:
As far as I know `Array` and `[]` does not have any difference in TS. You sometimes use one and sometimes the other in this interface. If they were consistent it would look better.